folding/src/expressions.rs

```diff
@@ -56,6 +57,67 @@ pub enum FoldingCompatibleExpr<C: FoldingConfig> {
     Pow(Box<Self>, u64),
 }
 
+impl<C: FoldingConfig> ToString for FoldingCompatibleExpr<C> {
+    fn to_string(&self) -> String {
+        match self {
+            FoldingCompatibleExpr::Atom(c) => match c {
+                FoldingCompatibleExprInner::Constant(c) => {
+                    let c = if c.is_zero() {
+                        "0".to_string()
+                    } else {
+                        c.to_string()
+                    };
+                    c.to_string()
+                }
+                FoldingCompatibleExprInner::Challenge(c) => {
+                    format!("{:?}", c)
+                }
+                FoldingCompatibleExprInner::Cell(cell) => {
+                    let Variable { col, row } = cell;
+                    let next = match row {
+                        CurrOrNext::Curr => "",
+                        CurrOrNext::Next => " * ω",
```

Comment:
This is technically incorrect: we're getting `col(x)` for `Curr` and `col(x * \omega)` for `Next`. Might be easier to write simply `Curr` or `Next`.